code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -42,116 +39,183 @@
 
-	if (!xeno.check_state())
-		return
+		return ..()
 
-	if (!isxeno_human(target) || xeno.can_not_harm(target))
-		to_chat(xeno, SPAN_XENOWARNING("We must target a hostile!"))
-		return
 
-	if (get_dist_sqrd(target, xeno) > 2)
-		to_chat(xeno, SPAN_XENOWARNING("[target] is too far away!"))
-		return
+/datum/action/xeno_action/onclick/feralrush/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/predatoralien = owner
 
-	var/mob/living/carbon/carbon = target
+	if (!action_cooldown_check())
+		return
 
-	if (carbon.stat == DEAD)
-		to_chat(xeno, SPAN_XENOWARNING("[carbon] is dead, why would we want to touch them?"))
+	if (!istype(predatoralien) || !predatoralien.check_state())
 		return
 
-	var/datum/behavior_delegate/predalien_base/behavior = xeno.behavior_delegate
-	if(!istype(behavior))
+	if (buffs_active)
+		to_chat(predatoralien, SPAN_XENOHIGHDANGER("We cannot stack this!"))
 		return
 
 	if (!check_and_use_plasma_owner())
 		return
 
-	ADD_TRAIT(carbon, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
 
-	if (ishuman(carbon))
-		var/mob/living/carbon/human/human = carbon
-		human.update_xeno_hostile_hud()
+	RegisterSignal(predatoralien, COMSIG_XENO_POST_TOUGHEN, PROC_REF(disable_feralrush))
+	addtimer(CALLBACK(src, PROC_REF(remove_effects)), duration)
 
+	predatoralien.add_filter("predalien_toughen", 1, list("type" = "outline", "color" = "#421313", "size" = 1))
+	predatoralien.balloon_alert(predatoralien, "we feel our muscles tense, making us faster and more durable for a time.", text_color = "#522020ff")
+	buffs_active = TRUE
+	predatoralien.speed_modifier -= speed_buff_amount
+	predatoralien.armor_modifier += armor_buff_amount
+	predatoralien.recalculate_speed()
+	predatoralien.recalculate_armor()
+	playsound(predatoralien, 'sound/voice/predalien_growl.ogg', 75, 0, status = 0)
 	apply_cooldown()
 
-	ADD_TRAIT(xeno, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
-	xeno.anchored = TRUE
 
-	if (do_after(xeno, activation_delay, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE))
-		xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] rips open the guts of [carbon]!"), SPAN_XENOHIGHDANGER("We rip open the guts of [carbon]!"))
-		carbon.spawn_gibs()
-		playsound(get_turf(carbon), 'sound/effects/gibbed.ogg', 75, 1)
-		carbon.apply_effect(get_xeno_stun_duration(carbon, 0.5), WEAKEN)
-		carbon.apply_armoured_damage(get_xeno_damage_slash(carbon, base_damage + damage_scale * behavior.kills), ARMOR_MELEE, BRUTE, "chest", 20)
+/datum/action/xeno_action/onclick/feralrush/proc/disable_feralrush()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/predatoralien = owner
+	if (buffs_active == TRUE)
+		predatoralien.balloon_alert(predatoralien, "our muscles relax, we no longer feel as armored.", text_color = "#522020ff")
+		predatoralien.remove_filter("predalien_toughen")
+	UnregisterSignal(predatoralien, COMSIG_XENO_POST_TOUGHEN)
+
+
+/datum/action/xeno_action/onclick/feralrush/proc/remove_effects()
```

Comment:
Combine this and `disable_ferralrush()` into one proc we don't need a whole other proc just to remove a filter.